src/main/java/picard/fastq/BamToBfq.java

```diff
@@ -1,108 +1,125 @@
-            READ_NAME_PREFIX = RUN_BARCODE + ":";
-        }
-        return null;
-    }
-
-}
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2009 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package picard.fastq;
+
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import picard.cmdline.CommandLineProgram;
+import picard.cmdline.StandardOptionDefinitions;
+import picard.cmdline.programgroups.SamOrBam;
+
+import java.io.File;
+
+/**
+ * Converts a BAM file into a BFQ (binary fastq formatted) file.
+ * <p>
+ * The BFQ format is useful or required to work with some tools like Maq aligner.
+ * </p>
+ * <h3>Input</h3>
+ * A single BAM file to convert.
+ * <h3>Output</h3>
+ * One or two FASTQ files depending on whether the BAM file contains data from a single or
+ * paired run. You must indicate the output directory that will contain these files (<code>ANALYSIS_DIR</code>)
+ * and the output file name prefix (<code>OUTPUT_FILE_PREFIX</code>). 
+ * <h3>Usage example:</h3>
+ * <pre>
+ *     java -jar picard.jar BamToBfq \
+ *             I=input.bam \
+ *             ANALYSIS_DIR=output_dir \
+ *             OUTPUT_FILE_PREFIX=output_name \
+ *             PAIRED_RUN=false
+ * </pre>
+ * 
+ * @author ktibbett@broadinstitute.org
+ */
+@CommandLineProgramProperties(
+        summary = BamToBfq.USAGE_SUMMARY + BamToBfq.USAGE_DETAILS,
+        oneLineSummary = BamToBfq.USAGE_SUMMARY,
+        programGroup = SamOrBam.class)
+@DocumentedFeature
+public class BamToBfq extends CommandLineProgram {
+    static final String USAGE_SUMMARY = "Create BFQ files from a BAM file for use by the maq aligner.  ";
+    static final String USAGE_DETAILS = "BFQ is a binary version of the FASTQ file format. This tool creates bfq files from a BAM file " +
```

Comment:
@vdauwera In most of the Picard tools, there is a USAGE_DETAILS string.

Should this string be exactly the same as what is in the javadoc?  I'm not sure I like that, because it smells like code duplication, but it's not immediately clear to me how these should differ.